npm start fails

Hey @pcarranzav, I just tried this out, clone and pushed on first container load and npm start fails. Am I missing something? or has something broken? 

```
18.08.15 16:08:41 [+0100] events.js:196
18.08.15 16:08:41 [+0100] throw TypeError('listener must be a function');
18.08.15 16:08:41 [+0100] ^
18.08.15 16:08:41 [+0100] TypeError: listener must be a function
18.08.15 16:08:41 [+0100] at TypeError (<anonymous>)
18.08.15 16:08:41 [+0100] at removeListener (events.js:196:11)
18.08.15 16:08:41 [+0100] at onChange (/usr/src/app/node_modules/connman-simplified/lib/WiFi.js:250:32)
18.08.15 16:08:41 [+0100] at emit (events.js:98:17)
18.08.15 16:08:41 [+0100] at null.<anonymous> (/usr/src/app/node_modules/connman-simplified/node_modules/connman-api/lib/service.js:87:12)
18.08.15 16:08:41 [+0100] at emit (events.js:117:20)
18.08.15 16:08:41 [+0100] at null.<anonymous> (/usr/src/app/node_modules/connman-simplified/node_modules/connman-api/node_modules/dbus/lib/bus.js:43:22)
18.08.15 16:08:41 [+0100] at emit (events.js:106:17)
18.08.15 16:08:41 [+0100] at /usr/src/app/node_modules/connman-simplified/node_modules/connman-api/node_modules/dbus/lib/dbus.js:70:42
```

@pcarranzav :)
```
18.08.15 16:58:15 [+0100]
18.08.15 16:58:15 [+0100] > resin-wifi-connect@1.0.0 start /usr/src/app
18.08.15 16:58:15 [+0100] > cd src && node app.js
18.08.15 16:58:15 [+0100]
18.08.15 16:58:18 [+0100] express:application set "x-powered-by" to true +0ms
18.08.15 16:58:18 [+0100] express:application set "etag" to 'weak' +26ms
18.08.15 16:58:18 [+0100] express:application set "etag fn" to [Function: wetag] +5ms
18.08.15 16:58:18 [+0100] express:application set "env" to 'development' +2ms
18.08.15 16:58:18 [+0100] express:application set "query parser" to 'extended' +2ms
18.08.15 16:58:18 [+0100] express:application set "query parser fn" to [Function: parseExtendedQueryString] +2ms
18.08.15 16:58:18 [+0100] express:application set "subdomain offset" to 2 +2ms
18.08.15 16:58:18 [+0100] express:application set "trust proxy" to false +1ms
18.08.15 16:58:18 [+0100] express:application set "trust proxy fn" to [Function: trustNone] +4ms
18.08.15 16:58:18 [+0100] express:application booting in development mode +2ms
18.08.15 16:58:18 [+0100] express:application set "view" to [Function: View] +1ms
18.08.15 16:58:18 [+0100] express:application set "views" to '/usr/src/app/src/views' +2ms
18.08.15 16:58:18 [+0100] express:application set "jsonp callback name" to 'callback' +1ms
18.08.15 16:58:18 [+0100] Starting node connman app
18.08.15 16:58:18 [+0100] Connman initialized
18.08.15 16:58:18 [+0100] WiFi initialized
18.08.15 16:58:18 [+0100] body-parser deprecated bodyParser: use individual json/urlencoded middlewares app.js:141:15
18.08.15 16:58:19 [+0100] body-parser deprecated undefined extended: provide extended option ../node_modules/body-parser/index.js:105:29
18.08.15 16:58:19 [+0100] express:router use / query +1s
18.08.15 16:58:19 [+0100] express:router:layer new / +3ms
18.08.15 16:58:19 [+0100] express:router use / expressInit +7ms
18.08.15 16:58:19 [+0100] express:router:layer new / +1ms
18.08.15 16:58:19 [+0100] express:router use / bodyParser +4ms
18.08.15 16:58:19 [+0100] express:router:layer new / +1ms
18.08.15 16:58:19 [+0100] express:router use / serveStatic +5ms
18.08.15 16:58:19 [+0100] express:router:layer new / +1ms
18.08.15 16:58:19 [+0100] express:router:route new /ssids +3ms
18.08.15 16:58:19 [+0100] express:router:layer new /ssids +2ms
18.08.15 16:58:19 [+0100] express:router:route get /ssids +3ms
18.08.15 16:58:19 [+0100] express:router:layer new / +1ms
18.08.15 16:58:19 [+0100] express:router:route new /connect +2ms
18.08.15 16:58:19 [+0100] express:router:layer new /connect +2ms
18.08.15 16:58:19 [+0100] express:router:route post /connect +1ms
18.08.15 16:58:19 [+0100] express:router:layer new / +1ms
18.08.15 16:58:19 [+0100] express:router use / <anonymous> +2ms
18.08.15 16:58:19 [+0100] express:router:layer new / +1ms
18.08.15 16:58:19 [+0100] Trying to join wifi
18.08.15 16:58:19 [+0100] connman-simplified:wifi joinFavorite +0ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi closeHotspot +10ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi [NOTE] Hotspot already closed +8ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi search service attempt: +5ms 1 / 5
18.08.15 16:58:19 [+0100] connman-simplified:wifi scan +2ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi join with agent: +35ms Techspace
18.08.15 16:58:19 [+0100] connman-simplified:wifi closeHotspot +1ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi [NOTE] Hotspot already closed +5ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi search service attempt: +4ms 1 / 5
18.08.15 16:58:19 [+0100] connman-simplified:wifi scan +2ms
18.08.15 16:58:19 [+0100] connman-simplified:wifi connect response: +25ms
18.08.15 16:58:19 [+0100]
18.08.15 16:58:19 [+0100] connman-simplified:wifi service State: +311ms association
18.08.15 16:58:55 [+0100] connman-simplified:base [ERROR] service error: +36s invalid-key
18.08.15 16:58:55 [+0100] connman-simplified:wifi service State: +2ms failure
18.08.15 16:58:55 [+0100]
18.08.15 16:58:55 [+0100] events.js:196
18.08.15 16:58:55 [+0100] throw TypeError('listener must be a function');
18.08.15 16:58:55 [+0100] ^
18.08.15 16:58:55 [+0100] TypeError: listener must be a function
18.08.15 16:58:55 [+0100] at TypeError (<anonymous>)
18.08.15 16:58:55 [+0100] at removeListener (events.js:196:11)
18.08.15 16:58:55 [+0100] at onChange (/usr/src/app/node_modules/connman-simplified/lib/WiFi.js:250:32)
18.08.15 16:58:55 [+0100] at emit (events.js:98:17)
18.08.15 16:58:55 [+0100] at null.<anonymous> (/usr/src/app/node_modules/connman-simplified/node_modules/connman-api/lib/service.js:87:12)
18.08.15 16:58:55 [+0100] at emit (events.js:117:20)
18.08.15 16:58:55 [+0100] at null.<anonymous> (/usr/src/app/node_modules/connman-simplified/node_modules/connman-api/node_modules/dbus/lib/bus.js:43:22)
18.08.15 16:58:55 [+0100] at emit (events.js:106:17)
18.08.15 16:58:55 [+0100] at /usr/src/app/node_modules/connman-simplified/node_modules/connman-api/node_modules/dbus/lib/dbus.js:70:42
18.08.15 16:58:55 [+0100]
18.08.15 16:58:56 [+0100] npm ERR! Linux 3.18.5
18.08.15 16:58:56 [+0100] npm ERR! argv "node" "/usr/local/bin/npm" "start"
18.08.15 16:58:56 [+0100] npm ERR! node v0.10.38
18.08.15 16:58:56 [+0100] npm ERR! npm v2.10.1
18.08.15 16:58:56 [+0100] npm ERR! code ELIFECYCLE
18.08.15 16:58:56 [+0100] npm ERR! resin-wifi-connect@1.0.0 start: `cd src && node app.js`
18.08.15 16:58:56 [+0100] npm ERR! Exit status 8
18.08.15 16:58:56 [+0100] npm ERR!
18.08.15 16:58:56 [+0100] npm ERR! Failed at the resin-wifi-connect@1.0.0 start script 'cd src && node app.js'.
18.08.15 16:58:56 [+0100] npm ERR! This is most likely a problem with the resin-wifi-connect package,
18.08.15 16:58:56 [+0100] npm ERR! not with npm itself.
18.08.15 16:58:56 [+0100] npm ERR! Tell the author that this fails on your system:
18.08.15 16:58:56 [+0100] npm ERR! cd src && node app.js
18.08.15 16:58:56 [+0100] npm ERR! You can get their info via:
18.08.15 16:58:56 [+0100] npm ERR! npm owner ls resin-wifi-connect
18.08.15 16:58:56 [+0100] npm ERR! There is likely additional logging output above.
18.08.15 16:58:56 [+0100]
18.08.15 16:58:56 [+0100] npm ERR! Please include the following file with any support request:
18.08.15 16:58:56 [+0100] npm ERR! /usr/src/app/npm-debug.log
18.08.15 16:58:56 [+0100] App exited but container is still running to allow web terminal access
```

--- a/src/app.coffee
+++ b/src/app.coffee
@@ -138,13 +138,7 @@ manageConnection = (retryCallback) ->
 			iptables.flushAsync('nat', 'TETHER')
 		.then ->
 			if !properties.connected
-				console.log('Trying to join wifi')
-				wifi.joinFavoriteAsync()
-				.then ->
-					console.log('Joined! Exiting.')
-					retryCallback()
-				.catch (err) ->
-					connectOrStartServer(wifi, retryCallback)
+				connectOrStartServer(wifi, retryCallback)
 			else
 				console.log('Already connected')
 				retryCallback()

